webapp/src/components/modals/run_playbook_modal.tsx

```diff
@@ -0,0 +1,238 @@
+import React, {ComponentProps, useState, useEffect} from 'react';
+
+import {useIntl} from 'react-intl';
+import styled from 'styled-components';
+import {useSelector} from 'react-redux';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+
+import {Client4} from 'mattermost-redux/client';
+
+import Icon from '@mdi/react';
+import {mdiNotebookOutline, mdiAccountOutline} from '@mdi/js';
+
+import {GlobalState} from 'mattermost-redux/types/store';
+
+import {displayUsername, getFullName} from 'mattermost-redux/utils/user_utils';
+import {getTeammateNameDisplaySetting} from 'mattermost-redux/selectors/entities/preferences';
+
+import {usePlaybook} from 'src/hooks';
+import {BaseInput} from 'src/components/assets/inputs';
+import GenericModal, {InlineLabel, Description} from 'src/components/widgets/generic_modal';
+import {createPlaybookRun} from 'src/client';
+import {navigateToUrl} from 'src/browser_routing';
+
+const ID = 'playbooks_run_playbook_dialog';
+
+export const makeModalDefinition = (playbookId: string, description: string, teamId: string, teamName: string) => ({
+    modalId: ID,
+    dialogType: RunPlaybookModal,
+    dialogProps: {playbookId, description, teamId, teamName},
+});
+
+type Props = {
+    playbookId: string,
+    description: string,
+    teamId: string,
+    teamName: string
+} & Partial<ComponentProps<typeof GenericModal>>;
+
+const owner = 'Owner';
+const playbookRunDescription = 'A channel will be created with this name';
+
+const RunPlaybookModal = ({
+    playbookId,
+    description,
+    teamId,
+    teamName,
+    ...modalProps
+}: Props) => {
+    const {formatMessage} = useIntl();
+
+    const [showModal, setShowModal] = useState(true);
+    const [runName, setRunName] = useState('');
+    const currentUser = useSelector(getCurrentUser);
+    const teamnameNameDisplaySetting = useSelector<GlobalState, string | undefined>(getTeammateNameDisplaySetting) || '';
+
+    const playbookOwner = getFullName(currentUser) || displayUsername(currentUser, teamnameNameDisplaySetting);
+    const profileUri = Client4.getProfilePictureUrl(currentUser.id, currentUser.last_picture_update);
+    const playbook = usePlaybook(playbookId);
+
+    useEffect(() => {
+        if (playbook) {
+            setRunName(playbook.channel_name_template);
+        }
+    }, [playbook, playbook?.id]);
+
+    const onSubmit = () => {
+        const playbookRun = createPlaybookRun(playbookId, currentUser.id, teamId, runName, description);
+        playbookRun.then(async (newPlaybookRun) => {
+            modalProps.onHide?.();
+            const pathname = `/playbooks/runs/${newPlaybookRun.id}`;
+            const search = '';
+            navigateToUrl({pathname, search});
+        }).catch(() => {
+            // show error
+        });
```

Comment:
you could simplify this a bit this piece (must change the import to use navigateToPluginUrl):
```suggestion
        createPlaybookRun(playbookId, currentUser.id, teamId, runName, description)
            .then((newPlaybookRun) => {
                modalProps.onHide?.();
                navigateToPluginUrl(`/runs/${newPlaybookRun.id}`);
            }).catch(() => {
                // show error
            });
```